adapters/pulsepoint_test.go

```diff
@@ -0,0 +1,193 @@
+package adapters
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/prebid/openrtb"
+	"github.com/prebid/prebid-server/pbs"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+/**
+ * Verify adapter names are setup correctly.
+ */
+func TestPulsePointAdapterNames(t *testing.T) {
+	adapter := NewPulsePointAdapter(DefaultHTTPAdapterConfig, "http://localhost/bid", "http://localhost")
+	VerifyStringValue(adapter.Name(), "PulsePoint", t)
+	VerifyStringValue(adapter.FamilyName(), "pulsepoint", t)
+}
+
+/**
+ * Verifies the user sync parameters.
+ */
+func TestPulsePointUserSyncInfo(t *testing.T) {
+	adapter := NewPulsePointAdapter(DefaultHTTPAdapterConfig, "http://localhost/bid", "http://localhost")
+	VerifyStringValue(adapter.GetUsersyncInfo().Type, "redirect", t)
+	VerifyStringValue(adapter.GetUsersyncInfo().URL, "https://bh.contextweb.com/rtset?pid=-1&ev=1&rurl=http%3A%2F%2Flocalhost%2Fsetuid%3Fbidder%3Dpulsepoint%26uid%3D%25%25VGUID%25%25", t)
+}
+
+/**
+ * Test required parameters not sent
+ */
+func TestPulsePointRequiredBidParameters(t *testing.T) {
+	adapter := NewPulsePointAdapter(DefaultHTTPAdapterConfig, "http://localhost/bid", "http://localhost")
+	ctx := context.TODO()
+	req := SampleRequest()
+	bidder := SampleBidder()
+	// remove "ct" param and verify error message.
+	bidder.AdUnits[0].Params = json.RawMessage("{\"cp\": 2001, \"cf\": \"728X90\"}")
+	_, errTag := adapter.Call(ctx, &req, &bidder)
+	VerifyStringValue(errTag.Error(), "Missing TagId param ct", t)
+	// remove "cp" param and verify error message.
+	bidder.AdUnits[0].Params = json.RawMessage("{\"ct\": 1001, \"cf\": \"728X90\"}")
+	_, errPub := adapter.Call(ctx, &req, &bidder)
+	VerifyStringValue(errPub.Error(), "Missing PublisherId param cp", t)
+	// remove "cf" param and verify error message.
+	bidder.AdUnits[0].Params = json.RawMessage("{\"cp\": 2001, \"ct\": 1001}")
+	_, errSize := adapter.Call(ctx, &req, &bidder)
+	VerifyStringValue(errSize.Error(), "Missing AdSize param cf", t)
+}
+
+/**
+ * Verify the openrtb request sent to Pulsepoint endpoint.
+ * Ensure the ct, cp, cf params are transformed and sent alright.
+ */
+func TestPulsePointOpenRTBRequest(t *testing.T) {
+	var ortbRequest openrtb.BidRequest
+	server := httptest.NewServer(
+		http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			// Send 204 (no-bid)
+			decoder := json.NewDecoder(req.Body)
+			decoder.Decode(&ortbRequest)
+			defer req.Body.Close()
+			w.WriteHeader(204)
+		}),
+	)
+	defer server.Close()
+	ctx := context.TODO()
+	req := SampleRequest()
+	bidder := SampleBidder()
```

Comment:
constructing the bidder by using the overall pbs request format (see adapters/facebook_test.go) is probably more resilient to changes in the PBSBidder internals. You already have the SampleRequest(), so if you call ParsePBSRequest() you can kill two birds with one stone.